docs/guides/app-payments/build-on-mobile.mdx

```diff
@@ -21,46 +19,199 @@ This page details how MobilePay app payments should be implemented in your Andro
+    try {
+        Intent intent = new Intent(Intent.ACTION_VIEW);
+        intent.setData(Uri.parse(deepLink));
+        startActivityForResult(intent, MOBILEPAY_PAYMENT_REQUEST_CODE);
+    } catch (android.content.ActivityNotFoundException exception) {
+        // Inform the user that MobilePay is not installed.
+    } catch (NullPointerException exception) {
+        // Null deepLink was passed in.
+    }
+}
+
+@Override
+protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+    super.onActivityResult(requestCode, resultCode, data);
+
+    if (requestCode == MOBILEPAY_PAYMENT_REQUEST_CODE) {
+        if (resultCode == SUCCESS_RESULT_CODE) {
+          
+        } else if (resultCode == REJECT_RESULT_CODE) {
+            
+        } else if (resultCode == FAILED_RESULT_CODE) {
+            
+        } else if (resultCode == UNKNOWN_RESULT_CODE) {
+            
+        }
+    }
+}
+```
+</TabItem>
+
 </Tabs>
 
-:::caution Make sure you are compliant with mobile application distribution service rules!
-Using app payments for digital sales might not be allowed by some mobile application distribution services (Google Play, App Store, etc.) and might result in your application being removed. Please review the terms of service for the respective mobile application distribution services you use to ensure that you are compliant with their guidelines.
-:::
+## Handling the Result
 
-## Create Payment
+MobilePay can return these result codes:
+  - <b> SUCCESS_RESULT_CODE</b> (Activity.RESULT_OK) - payment was successful and it is in reserved state.
+  - <b>REJECT_RESULT_CODE</b> (Activity.RESULT_CANCELED) - user rejected the payment.
+  - <b>FAILED_RESULT_CODE</b> (1) - user could not complete the payment due to invalid state of payment or his MobilePay account.
+  - <b>UNKNOWN_RESULT_CODE</b> (2) - user attempted to do a payment, but because of network issues we could not determine if it was successful. 
+  (E.g. User tries to do a payment, but it does not succeed due to network issues. Then he decides to reject the payment and again he receives a network error. 
+  At this point we cannot leave him blocked in MobilePay app, so we finish with this result code.) 
+  You should try to capture or at least check the state of these payments as they can be successful.
 
-First, you'll need to create payment. To do this you should call your backend service that is implemented as described [here](/docs/payments-refunds/create-payments). This service call should return a response that contains a deep link needed to launch the MobilePay app.
+</TabItem>
 
-## Redirect to MobilePay
+<TabItem value="deep_link">
 
-Now that we have the deep link let's open MobilePay app.
+Deep link approach is used when MobilePay app is launched with `startActivity` function. When using this navigation method the MobilePay app will run in separate process and invoke your app with a deep link after payment flow is finished.
```

Comment:
```suggestion
Deep link approach is used when the MobilePay app is launched with the `startActivity` function. When using this navigation method, the MobilePay app will run in a separate process and invoke your app with a deep link after the payment flow is finished.
```

Also, did you mean startActivityForResult instead of startActivity?